### DISCUSSION TITLE
bull usage problem

### DISCUSSION THREAD

**Discussion by chenhuanguang** [OptimalBits/bull#2598] [Q&A]
1. I used kue before in my project, but now I want to use bull. The redis library I used before is "redis". Do I need to replace it with "ioredis" library? Is bull compatible?
2. Which one is recommended for bull and bullmq? The requirement of my project is to require higher stability

**Comment by manast** [comment]
Bull and BullMQ work with ioredis. You can pass own ioredis instances or a config option. If you are new to Bull you should use BullMQ: 
https://docs.bullmq.io/
